hivemq-edge/src/main/java/com/hivemq/bridge/BridgeService.java

```diff
@@ -108,36 +118,36 @@ public synchronized void updateBridges(final @NotNull List<MqttBridge> bridges)
             }
         });
 
-        newMapOfKnownBridges.forEach((bridgeId, bridge) -> {
+        toUpdate.forEach(bridgeId -> {
             final var active = activeBridgeNamesToClient.get(bridgeId);
+            final var newBridge = bridgeIdToConfig.get(bridgeId);
             if(active != null) {
-                if(active.bridge().equals(bridge)) {
-                    log.debug("Not restarting bridge {} because config is unchanged", bridgeId);
+                if(active.bridge().equals(newBridge)) {
+                    log.debug("Not restarting bridge {} because config  is unchanged", bridgeId);
                 } else {
                     log.info("Restarting bridge {} because config has changed", bridgeId);
+                    allKnownBridgeConfigs.remove(bridgeId);
+                    allKnownBridgeConfigs.put(bridgeId, newBridge);
```

Comment:
These 2 lines are not atomic. There is a possibility that the bridge is not found by another thread. Is this expected? Why not call put directly without remove?